claim-check-pattern/call-usage-app/src/test/java/com/iluwatar/claimcheckpattern/HttpResponseMessageMock.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * The MIT License
+ * Copyright © 2014-2021 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwatar.claimcheckpattern;
+
+import com.microsoft.azure.functions.HttpResponseMessage;
+import com.microsoft.azure.functions.HttpStatus;
+import com.microsoft.azure.functions.HttpStatusType;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The mock for HttpResponseMessage, can be used in unit tests to verify if the
+ * returned response by HTTP trigger function is correct or not.
+ */
+public class HttpResponseMessageMock implements HttpResponseMessage {
+    private int httpStatusCode;
+    private HttpStatusType httpStatus;
+    private Object body;
+    private Map<String, String> headers;
+
+    public HttpResponseMessageMock(HttpStatusType status, Map<String, String> headers, Object body) {
+        this.httpStatus = status;
+        this.httpStatusCode = status.value();
+        this.headers = headers;
+        this.body = body;
+    }
+
+    @Override
+    public HttpStatusType getStatus() {
+        return this.httpStatus;
+    }
+
+    @Override
+    public int getStatusCode() {
+        return httpStatusCode;
+    }
```

Comment:
You can use Lombok's @Getter annotation to get rid of the boilerplate. Check this throughout the example.

Reply:
@iluwatar This is a mock test class. These are not `getter-setter` methods but override methods. You can ignore these.